### DISCUSSION TITLE
Dynamic menu

### DISCUSSION THREAD

**Discussion by nafg** [polybar/polybar#2736] [Q&A] [answered]
Hi, is there any way to make menu items change over time?

If there's no better way I'm thinking to have a script modify the config file.

Relatedly, is `-r` something that's passed on startup that puts polybar into a "watch mode," or is it more like a separate command meant to run after config has changed to notify the running polybar, more analogous to say `gnome-shell --replace` or `nginx -s reload` so to speak?

**Comment by patrick96** [comment, marked as answer]
No, the menu is static.

`polybar -r` means that instance of polybar will restart everytime it notices that its config file changed ("watch mode").

**Comment by nafg** [comment]
Thanks. Meanwhile I went with Argos. But I guess with -r I could have a script regenerate the config file continuously? Or would that cause flicker or other issues?

**Reply by patrick96** [reply]
You will definitely have flickers. `-r` will completely restart the bar, destroying and recreating the window.

**Comment by nafg** [comment]
Ok.I guess what people do instead for dynamic menus is script rofi or
something. But then the menu doesn't appear by the mouse.

**Reply by patrick96** [reply]
Yeah I think most people use rofi. Also you can make it appear by the mouse: `rofi -monitor -3 ...`

**Reply by jrock2004** [reply]
Did you end up doing this? If so do you have example config?

**Comment by nafg** [comment]
Oh neat
